src/SDSS.jl

```diff
@@ -130,5 +135,454 @@ function load_stamp_catalog(cat_dir, stamp_id, blob; match_blob=false)
+ - rncol: The number of columns in an eigenimage.
+ - cmat: The coefficients of the weight polynomial (see below).
+
+The point spread function is represented as an rnrow x rncol image
+showing what a true point source would look as viewed through the optics.
+This image varies across the field, and to parameterize this variation,
+the PSF is represented as a linear combination of four "eigenimages",
+with weights that vary across the image.  See get_psf_at_point()
+for more details.
+""" ->
+function load_psf_data(field_dir, run_num, camcol_num, frame_num, b)
+    @assert 1 <= b <= 5
+    psf_filename = "$field_dir/psField-$run_num-$camcol_num-$frame_num.fit"
+    psf_fits = FITS(psf_filename)
+    psf_hdu = psf_fits[b + 1]
+
+    nrows = read_key(psf_hdu, "NAXIS2")[1]
+    nrow_b = read(psf_hdu, "nrow_b")[1]
+    ncol_b = read(psf_hdu, "ncol_b")[1]
+    rnrow = read(psf_hdu, "rnrow")[1]
+    rncol = read(psf_hdu, "rncol")[1]
+    cmat = convert(Array{Float64, 3}, read(psf_hdu, "c"))
+
+    # Some low-level FITSIO rigamarole until the high-level interface can
+    # read variable length binary rows.
+    
+    # Move to the appropriate header.
+    FITSIO.Libcfitsio.fits_movabs_hdu(psf_hdu.fitsfile, psf_hdu.ext)
+    rrows_colnum = FITSIO.Libcfitsio.fits_get_colnum(psf_hdu.fitsfile, "RROWS")
+    rrows = zeros(Float64, rnrow * rncol, nrows)
+    for rownum in 1:nrows
+        repeat, offset = FITSIO.Libcfitsio.fits_read_descriptll(psf_hdu.fitsfile, rrows_colnum, rownum)
+        @assert repeat[1] == rnrow * rncol
+        result = zeros(repeat[1])
+        FITSIO.Libcfitsio.fits_read_col(psf_hdu.fitsfile, rrows_colnum, rownum, 1, result)
+        rrows[:, rownum] = result
+    end
+    close(psf_fits)
+
+    # Only the first (nrow_b, ncol_b) submatrix of cmat is used for reasons obscure
+    # to the author.
+    rrows, rnrow, rncol, cmat[1:nrow_b, 1:ncol_b, :]
+end
+
+@doc """
+Evaluate a gmm object at the data points x_mat.
+""" ->
+function evaluate_gmm(gmm::GMM, x_mat::Array{Float64, 2})
+    post = gmmposterior(gmm, x_mat) 
+    exp(post[2]) * gmm.w;
+end
+
+
+@doc """
+Fit a mixture of 2d Gaussians to a PSF image (evaluated at a single point).
+
+Args:
+ - psf: An matrix image of the point spread function, e.g. as returned by get_psf_at_point.
+ - tol: The target mean sum of squared errors between the MVN fit and the raw psf.
+ - max_iter: The maximum number of EM steps to take.
+
+ Returns:
+  - A GMM object containing the fit
+
+ Use an EM algorithm to fit a mixture of three multivariate normals to the
+ point spread function.  Although the psf is continuous-valued, we use EM by
+ fitting as if we had gotten psf[x, y] data points at the image location [x, y].
+ As of writing weighted data matrices of this form were not supported in GaussianMixtures.
+
+ Naturally, the mixture only matches the psf up to scale.
+""" ->
+function fit_psf_gaussians(psf::Array{Float64, 2}; tol = 1e-9, max_iter = 500)
```

Comment:
We might want the code for fitting the Celeste PSF in a different module, leaving just stuff for loading SDSS-formatted files here. I'm not sure how much it matters though